### PR TITLE
chore: [LLK-64] OriginalUrl and other parameters aligned to standard

### PR DESCRIPTION
**Pull request by SabPicariello**
## Short description
OriginalUrl and other parameters aligned to standard

## List of changes proposed in this pull request
- ts/features/lollipop/httpSignature/__tests__/signature.test.ts: updated test.
- ts/features/lollipop/httpSignature/signature.ts: removed request-target; added original-url.
- ts/features/lollipop/httpSignature/types/SignatureComponents.ts: removed request-target; added original-url.
- ts/features/lollipop/index.ts: added functions to properly get the `SignatureComponents`.
- ts/features/lollipop/utils/fetch.ts: removed unused `originalUrl` from `toSignatureComponents` and `SignatureConfigForgeInput`.

## How to test
Run `signature.test.ts` and check if everything passes.

### CODE REVIEW

File: ts/features/lollipop/index.ts [hunk 1]
```diff
@@ -17,19 +17,49 @@ export type LollipopConfig = {
...
+  const normalizedUrl = new URL(url.href);
+  normalizedUrl.port = "";
+
+  const decodedPathname = decodeURIComponent(
+    normalizedUrl.pathname.substring(1)
+  );
+  const encodedPathname = encodeURIComponent(decodedPathname);
+
+  const uppercaseEncodedPathname = encodedPathname.replace(
+    /%[0-9a-f]{2}/gi,
```
Comment (grausof) [inline ts/features/lollipop/index.ts:52]:
```suggestion
    /%\w\w/g,
```

File: ts/features/lollipop/index.ts [hunk 2]
```diff
@@ -17,19 +17,49 @@ export type LollipopConfig = {
...
     scheme: inputUrl.protocol,
-    targetUri: `${inputUrl.protocol}://${inputUrl.hostname}/${originalUrl}`
+    targetUri: normalizeForTargetUri(inputUrl),
+    originalUrl: getOriginalUrl(inputUrl)
   };
 }
 
+export function getOriginalUrl(inputUrl: URLParse) {
+  const newUrl = new URL(inputUrl.toString());
+  if (newUrl.port === "80") {
```
[the remark this answers is not in the document]
Comment (SabPicariello) [reply ts/features/lollipop/index.ts:34]: We doesn't want to show the port, in the original URL parameter,  if its value is 80 or 443.
For the second one, it ignores automatically the port, for the first one, we have to manually remove it.

[the remark this answers is not in the document]
Comment (grausof) [reply ts/features/lollipop/index.ts:34]: Port is always ignored if it is the same as the protocol specified in url.
Screenshot 2023-03-14 alle 13 25 06

File: ts/features/lollipop/index.ts [hunk 3]
```diff
@@ -17,19 +17,49 @@ export type LollipopConfig = {
...
+    return new URL(newUrl.toString().replace(`:${newUrl.port}`, "")).toString();
+  }
+
+  return newUrl.toString();
+}
+
+/* eslint-disable functional/immutable-data */
+export function normalizeForTargetUri(url: URLParse): string {
```
[the remark this answers is not in the document]
Comment (francescopersico) [reply ts/features/lollipop/index.ts:38]: Can we add test cases for this method?

[the remark this answers is not in the document]
Comment (grausof) [reply ts/features/lollipop/index.ts:38]: I think the reference used is wrong. Lollipop is based on [httpbis v15](https://datatracker.ietf.org/doc/draft-ietf-httpbis-message-signatures/15/) and normalization is not required.
See https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-message-signatures-15#name-target-uri

**Review by grausof — CHANGES_REQUESTED** [review]
Use https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-message-signatures-15#name-target-uri

**Review by hevelius — APPROVED** [review]
LGTM

**Review by shadowsheep1 — COMMENTED** [review]
LGTM

**Pull request merged**